### PR TITLE
Add primary button

### LINKED ISSUE #1248: PrimaryButton
**Package** lib-react-components

**Is your feature request related to a problem? Please describe.**

#1199 Added a shared GoldButton, but a better variant would be a `PrimaryButton` component that takes a color prop that then loads a Grommet `Button` with the appropriate theme customization needed to render the correct fill colors in the light and dark themes. The current variants are: zooniverse brand teal (theme default), gold, green, Done & Talk blue.

### PR DESCRIPTION
**Pull request by rogerhutchings**
Adds a `PrimaryButton` component, which is designed to replace `GoldButton` and the various custom themed buttons in the project app / classifier. `PrimaryButton` has the same API is [Grommet's `Button`](https://v2.grommet.io/button), but takes a more generalised `color` prop, which can be `gold`, `blue`, `teal` or `green`.

Closes #1248 

- [Button style guide on InVision](https://projects.invisionapp.com/d/main?origin=v7#/projects/prototypes/12923997/assets/77727637?folderID=28561320)

# Review Checklist

## General

- [ ] Are the tests passing locally and on Travis?
- [ ] Is the documentation up to date?

## Components
- [ ] Has a storybook story been created or updated?
- [ ] Is the component accessible? 
  - [ ] Can it be used with a screen reader? [BBC guide to testing with VoiceOver](https://bbc.github.io/accessibility-news-and-you/accessibility-and-testing-with-voiceover-os.html)
  - [ ] Can it be used from the keyboard? [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - [ ] Is it passing accessibility checks in the storybook?

## Apps

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `yarn panic && yarn bootstrap` or `docker-compose up --build` and app works as expected?

## Publishing

- [ ] Is the changelog updated?
- [ ] Are the dependencies updated for apps and libraries that are using the newly published library?

## Post-merging

- [ ] Did the app deploy to https://frontend.preview.zooniverse.org/projects/:project-name/:owner or https://frontend.preview.zooniverse.org/about?
- [ ] Is the new feature working or bug now fixed?
  - [ ] Is there a Talk or blog post written to announce the new feature(s)?
- [ ] Is the design working across browsers (Firefox, Chrome, Edge, Safari) and mobile?
  - [ ] Is this approved by our designer?
- [ ] Is this ready for production deployment?

### CODE REVIEW

File: packages/lib-grommet-theme/src/index.js [hunk 1]
```diff
@@ -47,7 +48,8 @@ const neutralColors = [
   darkGreen, // neutral-1
   darkTeal, // neutral-2
   navy, // neutral-3
-  gold // neutral-4
+  gold, // neutral-4
+  blue // neutral-5
```
Comment (srallen) [inline packages/lib-grommet-theme/src/index.js]: Could you update the changelog for the unreleased section as well as the table list of colors in the readme?

**Comment by srallen** [comment]
Is this still WIP?

**Comment by rogerhutchings** [comment]
@beckyrother can I get a review on these button states, just for the gold theme? There's no complete list of states in any one place that I could find, and I can't remember if any of them have evolved since the InVision list was created.

If this is okay, I'll apply the same themes for the teal, blue and green versions.

Also, none of the original button designs have `box-shadow`s, which were added to the workflow selection buttons. Do you want to add those in?

## Light theme

### Default

Screenshot 2019-12-02 12 52 03

### Disabled

Screenshot 2019-12-02 12 52 13

### Hover

Screenshot 2019-12-02 12 52 29

### Active

Screenshot 2019-12-02 12 52 40

## Dark theme

### Default

Screenshot 2019-12-02 12 58 18

### Disabled

Screenshot 2019-12-02 12 58 26

### Hover

Screenshot 2019-12-02 12 58 42

### Active

Screenshot 2019-12-02 12 58 53

File: packages/lib-grommet-theme/src/index.js (hunk 1 above)
Comment (rogerhutchings) [reply packages/lib-grommet-theme/src/index.js]: Done

**Comment by beckyrother** [comment]
@srallen The primary button should be gold – it stands out better from most backgrounds. 
Where are there contrast issues? Just doing a really quick look on the site, looks like a lot of the gold buttons on the non-logged-in home page are gold with white text so those should be updated; are there more?

**Comment by srallen** [comment]
I've opened issue #1354

File: packages/lib-react-components/src/PrimaryButton/PrimaryButton.spec.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    const wrapper = render(
+      <Grommet>
+        <PrimaryButton label={LABEL} />
+      </Grommet>
+    )
+    expect(wrapper.text()).to.equal(LABEL)
+  })
+
+  it('should pass down extra props', function () {
+    const wrapper = render(
```
[the remark this answers is not in the document]
Comment (rogerhutchings) [reply packages/lib-react-components/src/PrimaryButton/PrimaryButton.spec.js]: Actually, that does make more sense in this situation; we don't need to test `Button` here

**Pull request merged**